### ISSUE TITLE
Show symbols in file (i.e. Outline)

### ISSUE THREAD

**Issue by GandalfSaxe** [James-Yu/LaTeX-Workshop#768]
### Description

LaTeX Workshop does not help VSCode recognize symbols, i.e. you can't jump to a section of text with the `Go To Symbol in File...` command (⌘⇧O). I suggest making this into symbols for easy navigation:

- `chapter`
- `section`
- `subsection`
- `subsubsection` 

## How symbols look in LaTeX currently 

<img width="820" alt="image" src="https://user-images.githubusercontent.com/12127864/44845937-336d6000-ac4f-11e8-9a7e-00119eb3a8b3.png">

It also means that the Outline pane is empty, which would otherwise be populated by the the list of symbols (i.e. sections):

<img width="247" alt="image" src="https://user-images.githubusercontent.com/12127864/44845992-5ac42d00-ac4f-11e8-8bbe-d3a01868b5ec.png">

## How symbols look in Markdown

As an example, here is how the symbol list looks for a typical Markdown file:
<img width="602" alt="image" src="https://user-images.githubusercontent.com/12127864/44846076-9e1e9b80-ac4f-11e8-8ca8-ea95f9715978.png">

And the Outline contains the same but collapsable:

<img width="242" alt="image" src="https://user-images.githubusercontent.com/12127864/44846092-aaa2f400-ac4f-11e8-8b69-87ba78a51a9a.png">

**Comment by GandalfSaxe** [comment]
Btw I do see there's a "LaTeX Outline", but it doesn't currently work; if I click to open it, it contains nothing:

<img width="367" alt="image" src="https://user-images.githubusercontent.com/12127864/44846520-dc688a80-ac50-11e8-8e62-a4dafc2cf6f9.png">

And also give this error:

<img width="455" alt="image" src="https://user-images.githubusercontent.com/12127864/44846553-f1ddb480-ac50-11e8-9b1b-da147c990019.png">

**Comment by jlelong** [comment]
Can you post a minimal example reproducing the _LaTeX Outline_ problem? I thought it had been fixed.

**Comment by GandalfSaxe** [comment]
Huh when I checked again it works now. I'll let you know if it happens again, and if I see a pattern in what could cause it.

Anyway my feature request of using the various section levels + chapters as a symbol list still stands, it'd be really useful 🙂

**Comment by werunom** [comment]
+1 vote for this. Similar request has been done [here](https://github.com/James-Yu/LaTeX-Workshop/issues/680).

@jlelong - the latex-outline problem fix (tracked [here](https://github.com/James-Yu/LaTeX-Workshop/issues/752)) probably has not yet been pushed. In my case, it is still not working. 
When the fix might be released?

**Comment by James-Yu** [comment]
Will be released shortly.

**Comment by James-Yu** [comment]
Sections worked originally. The new release is not about providing a new feature but refining it. If there is no outline shown for your project for now, pleae provide your source file since it is currently working for other projects.

**Comment by GandalfSaxe** [comment]
@James-Yu I want to distinguish between outline being available in the sidebar, which I suspect you may be talking about vs. outline being available in "Go to symbol in file" command (`workbench.action.gotoSymbol`), which is really what this issue was about.

**Comment by James-Yu** [comment]
Actually in either versions of the extension, the `outline` view can be populated by the LaTeX content. If it's empty, something wrong happened.

In the current version:

![image](https://user-images.githubusercontent.com/4210342/45265838-f6417300-b483-11e8-80b7-58506596a21c.png)

![image](https://user-images.githubusercontent.com/4210342/45265843-0c4f3380-b484-11e8-98ca-d39d46126501.png)

If the outlines above is not expected behavior, please give a more concrete example.

**Comment by GandalfSaxe** [comment]
Ok here are concrete examples 🙂 

## This is the outline you're talking about
<img width="1440" alt="screenshot 2018-09-09 at 17 02 20" src="https://user-images.githubusercontent.com/12127864/45265891-b7df9000-b452-11e8-91bf-6ba5ec7e2beb.png">

## This is the "current file symbol list" I'm talking about
<img width="1440" alt="screenshot 2018-09-09 at 17 03 22" src="https://user-images.githubusercontent.com/12127864/45265900-c7f76f80-b452-11e8-9a73-b9b123bd2866.png">

It's populated by equations and figures right now. What I would like to see is the same content of "Outline / LaTeX Outline" in this symbol list.

## What the symbol list looks like with markdown
<img width="1440" alt="screenshot 2018-09-09 at 17 04 34" src="https://user-images.githubusercontent.com/12127864/45265908-ef4e3c80-b452-11e8-9cdb-bb89efdf4f6e.png">

Basically I want the same of LaTeX

## The purpose of having sections in symbol list
..is to be able to quickly jump to a section using fuzzy search. Such is possible with Markdown in VSCode (using the "Markdown All in One" extension, or as demonstrated here for the markdown editor Caret:
![](https://i.imgur.com/kLdII9H.gif)


I think it'd be fine to have both figures, equations and outline in the symbols list, i.e. not have the outline replace, but added to the current list of symbols.

**Comment by James-Yu** [comment]
It is implemented in the current new version. An example:
![image](https://user-images.githubusercontent.com/4210342/45273325-c8404b00-b4e4-11e8-9d95-febe8eb26ec7.png)
It is unlikely that figures and tables will be included due to the complex syntax structures these two floats have.

**Comment by James-Yu** [comment]
Because regex seems not powerful enough to identify which environment the current \caption is in. I'll look into it, but it seems not quite likely.

**Comment by James-Yu** [comment]
This one should serve the purpose.

![image](https://user-images.githubusercontent.com/4210342/45465773-3c7b2880-b749-11e8-93b9-9b2940780d3b.png)

**Comment by GandalfSaxe** [comment]
Yes I think that’s good actually.

It’s strange the commands are called the same, with the distinction of being symbols in file and symbols in workspace. Obviously it’s also a different set of symbols l, not just a question of scope.

Anyway this issue is surely concluded now with a satisfying conclusion. Thanks for your work.
